packages/core/core/src/factories/useCartFactory.ts

```diff
@@ -43,55 +44,76 @@ export const useCartFactory = <CART, CART_ITEM, PRODUCT, COUPON>(
     const addItem = async ({ product, quantity, customQuery }) => {
       Logger.debug('useCart.addItem', { product, quantity });
 
-      loading.value = true;
-      const updatedCart = await factoryParams.addItem(
-        context,
-        {
-          currentCart: cart.value,
-          product,
-          quantity,
-          customQuery
-        }
-      );
-      cart.value = updatedCart;
-      loading.value = false;
+      try {
+        loading.value = true;
+        error.value.addItem = null;
+        const updatedCart = await factoryParams.addItem(
+          context,
+          {
+            currentCart: cart.value,
+            product,
+            quantity,
+            customQuery
+          }
+        );
+        cart.value = updatedCart;
+      } catch (err) {
+        error.value.addItem = err;
+        Logger.error('useCart/addItem', err);
+      } finally {
+        loading.value = false;
+      }
     };
 
     const removeItem = async ({ product, customQuery }) => {
       Logger.debug('useCart.removeItem', { product });
 
-      loading.value = true;
-      const updatedCart = await factoryParams.removeItem(
-        context,
-        {
-          currentCart: cart.value,
-          product,
-          customQuery
-        }
-      );
-      cart.value = updatedCart;
-      loading.value = false;
-    };
-
-    const updateItemQty = async ({ product, quantity, customQuery }) => {
-      Logger.debug('useCart.updateItemQty', { product, quantity });
-
-      if (quantity && quantity > 0) {
+      try {
         loading.value = true;
-        const updatedCart = await factoryParams.updateItemQty(
+        error.value.removeItem = null;
+        const updatedCart = await factoryParams.removeItem(
           context,
           {
             currentCart: cart.value,
             product,
-            quantity,
             customQuery
           }
         );
         cart.value = updatedCart;
+      } catch (err) {
+        error.value.removeItem = err;
+        Logger.error('useCart/removeItem', err);
+      } finally {
         loading.value = false;
       }
     };
 
+    const updateItemQty = async ({ product, quantity, customQuery }) => {
+      Logger.debug('useCart.updateItemQty', { product, quantity });
+
+      if (quantity && quantity > 0) {
```

Comment:
don't do it here, there will be some platforms where you won't even pass the quantity also it's something to handle on the UI

Reply:
We decided to leave it like it is for now.